### PR TITLE
fix: #288 Fixed Snippet tabs overwriting other tabs when closed

### PR DESCRIPTION
**Pull request by redcom**
This should fix the issue for you.

### CODE REVIEW

**Comment by asmsuechan** [comment]
@redcom Thanks so much for fixing! It almost behaves correct, but I find an error when I delete the rightmost tab.

f8523a4e2d9ab48e7c0e3c1e1255013b

```
SnippetNoteDetail.js?64bd:267 Uncaught TypeError: Cannot read property 'reload' of undefined
```

**Comment by redcom** [comment]
This is true. I'll have a look later today.

**Comment by asmsuechan** [comment]
Thank you!

**Comment by redcom** [comment]
Coll. I submitted a fix. It should be good now. Can you tell me what do you use to produce those animated gifs that show the issue?

**Comment by asmsuechan** [comment]
@redcom This works fine. Thank you! I use [Gyazo](https://gyazo.com/en) to capture my screen :smile: It can take at most 6 sec.

BTW, you **DON'T** need to combine your commits into one. Because it's trouble when I rebase your branch.

File: browser/main/Detail/SnippetNoteDetail.js [hunk 1]
```diff
@@ -250,17 +250,21 @@ class SnippetNoteDetail extends React.Component {
...
-    snippets.splice(index, 1)
-    this.state.note.snippets = snippets
-    let snippetIndex = this.state.snippetIndex >= snippets.length
-      ? snippets.length - 1
-      : this.state.snippetIndex
-    this.setState({
-      note: this.state.note,
-      snippetIndex
+    this.setState(() => {
+      let snippets = this.state.note.snippets.slice()
```
Comment (asmsuechan) [inline browser/main/Detail/SnippetNoteDetail.js]: Could you change `let` to `const`?

**Comment by redcom** [comment]
sure

File: browser/main/Detail/SnippetNoteDetail.js (hunk 1 above)
Comment (redcom) [inline browser/main/Detail/SnippetNoteDetail.js]: sure

**Comment by asmsuechan** [comment]
I assume this is more readable.

```
  deleteSnippetByIndex (index) {
    const snippets = this.state.note.snippets.slice()
    snippets.splice(index, 1)
    const note = Object.assign({}, this.state.note, {snippets})
    const snippetIndex = this.state.snippetIndex >= snippets.length
      ? snippets.length - 1
      : this.state.snippetIndex
    this.setState({ note, snippetIndex }, () => {
      this.save()
      this.refs['code-' + this.state.snippetIndex].reload()
    })
  }
```

**Comment by redcom** [comment]
It looks like so.

**Comment by asmsuechan** [comment]
Cool! LGTM

**Comment by redcom** [comment]
Thanks for accepting my commiting

**Pull request merged**